mindmeld/components/entity_resolver.py

```diff
@@ -536,17 +691,76 @@ def _construct_whitelist_query(entity, weight=1, use_phons=False):
 
                 results.append(result)
 
-            return results[0:20]
+            return results
+
+    def _load(self):
+        """Loads the trained entity resolution model from disk."""
+        try:
+            scoped_index_name = get_scoped_index_name(
+                self._app_namespace, self._es_index_name
+            )
+            if not self._es_client.indices.exists(index=scoped_index_name):
+                self.fit()
+        except EsConnectionError as e:
+            logger.error(
+                "Unable to connect to Elasticsearch: %s details: %s", e.error, e.info
+            )
+            raise EntityResolverConnectionError(es_host=self._es_client.transport.hosts) from e
+        except TransportError as e:
+            logger.error(
+                "Unexpected error occurred when sending requests to Elasticsearch: %s "
+                "Status code: %s details: %s",
+                e.error,
+                e.status_code,
+                e.info,
+            )
+            raise EntityResolverError from e
+        except ElasticsearchException as e:
+            raise EntityResolverError from e
+
 
-    def _predict_exact_match(self, entity):
-        """Predicts the resolved value(s) for the given entity using the loaded entity map.
+class ExactMatchEntityResolver(EntityResolverModelBase):
+    """
+    Resolver class based on exact matching
+    """
 
-        Args:
-            entity (Entity): An entity found in an input query
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **_kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config)
+        self._entity_mapping = None
+
+    @property
+    def default_er_config(self):
+        defaults = {
+            "model_settings": {
+                "augment_lower_case": False
+            }
+        }
+        return defaults
+
+    def _fit(self, clean):
+        """Loads an entity mapping file to resolve entities using exact match.
+        """
+        if clean:
+            logger.info(
+                "clean=True ignored while fitting exact_match algo for entity resolution"
+            )
+
+        entity_map = self.entity_map
+        augment_lower_case = self.er_config["model_settings"]["augment_lower_case"]
+        self._entity_mapping = self._process_entity_map(
+            self.type, entity_map, normalizer=self.normalizer,
+            augment_lower_case=augment_lower_case
+        )
+
+    def _predict(self, nbest_entities, top_n):
```

Comment:
Why doesnt top_n have a default, like `n=10`?

Reply:
The `predict()` method has one. So it is passed to `_predict()`.